app/javascript/src/Onboarding.jsx

```diff
@@ -173,16 +179,15 @@ class Onboarding extends Component {
     })
       .then(response =>
         response.json().then(json => {
-          this.setState({
-            allTags: this.state.allTags.map(currentTag => {
+          this.setState(prevState => ({
+            allTags: prevState.allTags.map(currentTag => {
               const newTag = currentTag;
-              if (currentTag.name === tag.name) {
-                newTag.following = json.outcome === 'followed';
-              }
+              newTag.following =
+                currentTag.name === tag.name ? json.outcome : 'followed';
```

Comment:
The functionality here changed. `newTag`'s `following` should be a be a boolean and not a string.